source/Cosmos.System2/Network/IPv4/TCP/Tcp.cs

```diff
@@ -141,6 +141,40 @@ public class TransmissionControlBlock
     /// </remarks>
     public class Tcp
     {
+
+        public static ushort DynamicPortStart = 49152;
+
+        private static Random dynamicPortStartRandom = new Random();
+
+        /// <summary>
+        /// gets a random port
+        /// </summary>
+        /// <param name="tries"></param>
+        /// <returns></returns>
```

Comment:
Random.Next can throw exception.
Maybe it's better to just take (the last used port +1)? you won't need tries parameter any more and it would improve the speed of the process + time complexity.

I also think this function should use mutex to prevent race conditions. In multithread system the CPU can do task switch on the last "if" and the kernel will give the same port to 2 different process at the same time.

Reply:
cosmos does not have at form of mutex right now